File: Tutorials/Parallelisation/code/to_order/serial.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>
#include <time.h>

#define SIZE 4
#define MAXRAND 10000

void rand_vec(int* random){
  int i;
  for(i=0; i<SIZE; i++) {
      random[i] = rand() % MAXRAND; 
  }
}

void cmp_vec(int* random, int* is_sorted){
  int i;
  for(i=1; i<SIZE; i++) {
    *is_sorted = *is_sorted + (random[i-1] < random[i]);
  }
}

int main(int argc, char** argv) {

  int random[SIZE];
  int is_sorted;

  srand(time(NULL));   // init random generator
 
  is_sorted = 0;

  rand_vec(random);
  cmp_vec(random,&is_sorted);

  printf("%d neighbouring sorted pairs found!\n",is_sorted);

}

File: Tutorials/Parallelisation/code/to_order/serial.py
import time
import random

SIZE = 2
MAXRAND = 10000

def rand_vec():

   random_ = []

   for i in range(SIZE):
      random_.append(random.randint(0,MAXRAND-1))

   return random_

def cmp_vec(random):
   
   is_sorted = 0
   last_value = None
   for value in random:
      if last_value is not None:
         is_sorted = is_sorted + (last_value < value)
      last_value = value

   return is_sorted


if __name__ == '__main__':

   random.seed(int(time.time())) # init random generator

   random_ = rand_vec()
   is_sorted = cmp_vec(random_)

   print("%d neighbouring sorted pairs found!" % (is_sorted))




